### PR TITLE
quiche.h: ensure size_t and ssize_t are defined

### PR DESCRIPTION
**Pull request by Lekensteyn**
Ensure that check_symbol_exists calls in CMake do not fail due to
missing headers. The minimal reproducer can be stripped down to:

    echo '#include <quiche.h>' | gcc -Iinclude -fsyntax-only -xc -

A similar (untested) fix for Windows was taken from
https://github.com/cloudflare/quiche/issues/450

### CODE REVIEW

**Comment by Lekensteyn** [comment]
The examples in `examples/` might not be directly usable with Windows, and the Makefile will definitely need changes. I can provide a tiny CMake check to catch regressions on this fix, that will automatically work for all platform (Linux, macOS, Windows).

Alternatively, what do you think of a CMakeLists.txt file in the `examples` directory that replaces the current examples/Makefile?

**Comment by junhochoi** [comment]
> Alternatively, what do you think of a CMakeLists.txt file in the examples directory that replaces the current examples/Makefile?

👍

File: include/quiche.h [hunk 1]
```diff
@@ -33,6 +33,14 @@ extern "C" {
 
 #include <stdint.h>
 #include <stdbool.h>
+#include <stddef.h>
+#ifdef __unix__
+#include <sys/types.h>
```
Comment (ghedo) [inline include/quiche.h:38]: Why is this needed? Shouldn't `stddef.h` be enough?

Comment (Lekensteyn) [reply include/quiche.h:38]: `size_t` is part of the C standard, but `ssize_t` is part of POSIX
http://man7.org/linux/man-pages/man0/stddef.h.0p.html
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_types.h.html

For POSIX systems inclusion of sys/types.h without stddef.h might be sufficient, but that might not work for Windows.

**Comment by ghedo** [comment]
Merged, thanks @Lekensteyn.
